gatling-jms/src/main/scala/io/gatling/jms/action/JmsRequestSend.scala

```diff
@@ -0,0 +1,144 @@
+ */
+object JmsRequestSend extends NameGen {
+
+  def apply(attributes: JmsAttributes, protocol: JmsProtocol, system: ActorSystem, statsEngine: StatsEngine, next: Action) = {
+    val actor = system.actorOf(JmsRequestSendActor.props(attributes, protocol, statsEngine, next))
+    new ExitableActorDelegatingAction(genName("jmsRequest"), statsEngine, next, actor)
+  }
+}
+
+object JmsRequestSendActor {
+
+  def props(attributes: JmsAttributes, protocol: JmsProtocol, statsEngine: StatsEngine, next: Action) =
+    Props(new JmsRequestSendActor(attributes, protocol, statsEngine, next))
+}
+
+class JmsRequestSendActor(attributes: JmsAttributes, protocol: JmsProtocol, val statsEngine: StatsEngine, val next: Action)
+    extends ValidatedActionActor {
+
+  // Create a client to refer to
+  val client = JmsClient(protocol, attributes.destination, attributes.replyDestination)
+
+  /**
+   * Framework calls the execute() method to send a single request
+   * <p>
+   * Note this does not catch any exceptions (even JMSException) as generally these indicate a
+   * configuration failure that is unlikely to be addressed by retrying with another message
+   */
+  def executeOrFail(session: Session): Validation[Unit] = {
+
+    val messageProperties = resolveProperties(attributes.messageProperties, session)
+
+    val jmsType = resolveOptionalExpression(attributes.jmsType, session)
+
+    // send the message
+    val startDate = nowMillis
+
+    val msg = jmsType.flatMap(jmsType =>
+      messageProperties.flatMap { props =>
+        attributes.message match {
+          case BytesJmsMessage(bytes) => bytes(session).map(bytes => client.sendBytesMessage(bytes, props, jmsType))
+          case MapJmsMessage(map) => map(session).map(map => client.sendMapMessage(map, props, jmsType))
+          case ObjectJmsMessage(o) => o(session).map(o => client.sendObjectMessage(o, props, jmsType))
+          case TextJmsMessage(txt) => txt(session).map(txt => client.sendTextMessage(txt, props, jmsType))
+        }
+      })
+    val endDate = nowMillis
+
+    msg.map { msg =>
+      processMessage(session, startDate, endDate, attributes.checks, msg, next, attributes.requestName);
+      // notify the tracker that a message was sent
+      //      val matchId = messageMatcher.requestMatchId(msg)
+      //      logMessage(s"Message sent JMSMessageID=${msg.getJMSMessageID} matchId=$matchId", msg)
+      //      tracker ! MessageSent(replyDestinationName, matchId, startDate, attributes.checks, session, next, attributes.requestName)
+    }
+
+  }
+
+  /**
+   * Processes a matched message
+   */
+  private def processMessage(
+    session: Session,
+    sent: Long,
+    received: Long,
+    checks: List[JmsCheck],
+    message: Message,
+    next: Action,
+    title: String): Unit = {
+
+    // run all the checks, advise the Gatling API that it is complete and move to next
+    val (checkSaveUpdate, error) = Check.check(message, session, checks)
```

Comment:
How can we have checks, as we're just in fire-and-forget mode? How would performing checks on sent messages be useful? Please explain.